common/utils.js

```diff
@@ -172,18 +172,42 @@
         notNull: "<i>All Records With Value</i>"
     })
 
-    .factory('UriUtils', ['$injector', '$rootScope', '$window', 'appContextMapping', 'appTagMapping', 'ContextUtils', 'Errors', 'messageMap', 'parsedFilter',
-        function($injector, $rootScope, $window, appContextMapping, appTagMapping, ContextUtils, Errors, messageMap, ParsedFilter) {
-        var chaiseConfig = Object.assign({}, $rootScope.chaiseConfig);
+    .factory('UriUtils', ['appContextMapping', 'appTagMapping', 'ConfigUtils', 'ContextUtils', 'defaultChaiseConfig', 'Errors', 'messageMap', 'parsedFilter', '$injector', '$rootScope', '$window',
+        function(appContextMapping, appTagMapping, ConfigUtils, ContextUtils, defaultChaiseConfig, Errors, messageMap, ParsedFilter, $injector, $rootScope, $window) {
+        var chaiseBaseURL;
+
+        function getCatalogId() {
+            var chaiseConfig = ConfigUtils.getConfigJSON();
+
+            if ($rootScope.context && $rootScope.context.catalogID) {
+                return "" + $rootScope.context.catalogID;
+            }
+
+            // NOTE: what if it's '?' instead of '#'
+            if ($window.location.hash != "") {
+                return getCatalogIDFromLocation();
```

Comment:
Change `getCatalogIDFromLocation` to return false or null if the hash doesn't exist at all (instead of throwing the error, you should just catch and return false/null). Then in here, you can check for the string and use it.